### PR TITLE
Integrate DAG tool

### PR DESCRIPTION
**Pull request by jlingema**
- Provides a class that can build [DAG](https://github.com/HEP-FCC/dag) from an `fcc::MCParticleCollection`

### CODE REVIEW

**Comment by jlingema** [comment]
Ready to merge from my point of view.

File: utilities/ParticleUtils.cc [hunk 1]
```diff
@@ -8,6 +8,7 @@
 // datamodel
```
Comment (vvolkl) [inline utilities/ParticleUtils.cc:8]: Could you remove the commented code in this file?

File: examples/graph.cc [hunk 2]
```diff
@@ -0,0 +1,44 @@
+#include "dag/DirectedAcyclicGraph.h"
+
+/** @file graph.cc
+  *
+  * This is a very simple example for using the nodes and visitors defined by DirectedAcyclicGraph.
+  * The idpair is very similar to the PODIO ObjectID, so this may be instructive. For a more complete
+  * example on how to use it in the context of the fcc-edm, have a look at the read example in fcc-physics.
```
Comment (vvolkl) [inline examples/graph.cc]: I may have missed something, but the read example in fcc-physics doesn't use graphs?

File: utilities/ParticleGraph.cc [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  fcc::ParticleGraph::VertexMap endVertexMap;
+
+  for (const auto& mcpart : particles) {
+    auto& startID = mcpart.startVertex().getObjectID();
+    auto& endID = mcpart.endVertex().getObjectID();
+    auto siblingsIt = startVertexMap.find(startID.index);
+    auto parentsIt = endVertexMap.find(endID.index);
+
+    unsigned particleIdx = mcpart.getObjectID().index;
+    if (startID.collectionID != -2) {
```
Comment (vvolkl) [inline utilities/ParticleGraph.cc]: Could you add a comment what -2 is in this context?

**Review by vvolkl — CHANGES_REQUESTED** [review]

**Comment by vvolkl** [comment]
Thanks, looks great! Maybe you can remove the commented code in the file `ParticleUtils.cc`, and I noted the few places where things weren't totally clear to me.

**Comment by jlingema** [comment]
Hi Valentin,
Thanks for the feedback. Your comments are addressed in the last commits. Instead of referring to fcc-physics for an example, I added a simple test (actually before we were missing any test of that class) here that uses the `ParticleGraph` and refer to that. It shows at least how the basic interaction works. The actual implementation in FCCSW will also be an example for more refined usage and is referred to (will create a PR there, once this is accepted and installed).

Cheers,
Joschka

**Review by vvolkl — APPROVED** [review]

**Pull request merged**